ManagedClientConsoleAppSample/Program.cs

```diff
@@ -1,68 +1,81 @@
-        internal const string clientId = "872cd9fa-d31f-45e0-9eab-6e460a02d1f1";          //change to your app registration's Application ID, unless you are an MSA backed account
-        internal const string replyUri = "urn:ietf:wg:oauth:2.0:oob";                     //change to your app registration's reply URI, unless you are an MSA backed account
-        //==========================================================================
+        //
+        // The Client ID is used by the application to uniquely identify itself to Azure AD.
+        // The Tenant is the name or Id of the Azure AD tenant in which this application is registered.
+        // The AAD Instance is the instance of Azure, for example public Azure or Azure China.
+        // The Authority is the sign-in URL of the tenant.
+        //
+        internal static string aadInstance = ConfigurationManager.AppSettings["ida:AADInstance"];
+        internal static string tenant = ConfigurationManager.AppSettings["ida:Tenant"];
+        internal static string clientId = ConfigurationManager.AppSettings["ida:ClientId"];
+        internal static string authority = String.Format(CultureInfo.InvariantCulture, aadInstance, tenant);
+
+        //URL of your Azure DevOps account.
+        internal static string azureDevOpsOrganizationUrl = ConfigurationManager.AppSettings["ado:OrganizationUrl"];
 
-        internal const string azureDevOpsResourceId = "499b84ac-1321-427f-aa17-267ca6975798"; //Constant value to target Azure DevOps. Do not change  
+        internal static string[] scopes = new string[] { "499b84ac-1321-427f-aa17-267ca6975798/user_impersonation" }; //Constant value to target Azure DevOps. Do not change  
+        
+        // MSAL Public client app
+        private static IPublicClientApplication application;
 
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            AuthenticationContext ctx = GetAuthenticationContext(null);
+            var accessToken = await SignInUserAndGetTokenUsingMSAL(scopes);
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", accessToken);
+            
+            ListProjects(bearerAuthHeader);
+
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Sign-in user using MSAL and obtain an access token for Azure DevOps
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns>Access Token</returns>
+        private static async Task<string> SignInUserAndGetTokenUsingMSAL(string[] scopes)
+        {
+            // Initialize the MSAL library by building a public client application
+            application = PublicClientApplicationBuilder.Create(clientId)
+                                       .WithAuthority(authority)
+                                       .WithDefaultRedirectUri()
+                                       .Build();
+            
```

Comment:
You might want to add token cache serialization.:
https://github.com/AzureAD/microsoft-identity-web/blob/b15e6a216805f4bee77894915b0f06599d0651e4/tools/app-provisioning-tool/app-provisioning-lib/DeveloperCredentials/MsalTokenCredential.cs#L49

and 

https://github.com/AzureAD/microsoft-identity-web/blob/b15e6a216805f4bee77894915b0f06599d0651e4/tools/app-provisioning-tool/app-provisioning-lib/DeveloperCredentials/MsalTokenCredential.cs#L71-L72